capsules/classifier_safety_gear_openvino/backend.py

```diff
@@ -0,0 +1,85 @@
+from typing import Dict, Union
+
+import numpy as np
+
+from vcap import (
+    DETECTION_NODE_TYPE,
+    OPTION_TYPE,
+    BaseStreamState,
+)
+
+from vcap_utils import (
+    BaseOpenVINOBackend,
+    linear_assignment,
+    iou_cost_matrix,
+)
+
+from .config import (
+    safety_hat,
+    safety_vest,
+    safety_gears,
+    attributes,
+)
+
+from . import config
+
+Num = Union[int, float]
+
+
+class Backend(BaseOpenVINOBackend):
+    label_map: Dict[int, str] = {1: "unknown", 2: safety_vest,
+                                 3: "unknown", 4: safety_hat}
+
+    def process_frame(self, frame: np.ndarray,
+                      detection_nodes: DETECTION_NODE_TYPE,
+                      options: Dict[str, OPTION_TYPE],
+                      state: BaseStreamState) -> DETECTION_NODE_TYPE:
+        if len(detection_nodes) == 0:
+            return detection_nodes
+
+        confidence_threshold = options[config.confidence_threshold]
+
+        input_dict, resize = self.prepare_inputs(frame)
+        prediction = self.send_to_batch(input_dict).get()
+        detections = self.parse_detection_results(
+            prediction, resize, self.label_map,
+            min_confidence=confidence_threshold)
+
+        safety_gear_detections = {}
+        for gear_type in safety_gears:
```

Comment:
Would you mind reusing code a bit more by splitting out the for loops in line 49 and 54 into a function. 

Something like this:
```python3

def assign_gear_attributes(person_detections, gear_detections, gear_type):
    """This function assigns DetectionNode.attributes[gear_type] to person
    detections, based on the location of gear detection bounding boxes.
    """
    gear_detections = [g for g in gear_detections if g.class_name == gear_type]

    for person_det in person_detections:
        gear_value = attributes[gear_type]["possible_values"][0]
        person_det.attributes[attributes[gear_type]["attribute_name"]] = gear_value
        person_det.extra_data[attributes[gear_type]["iou"]] = 0
        person_det.extra_data[attributes[gear_type]["confidence"]] = 0

    if len(gear_detections) == 0:
        return

    iou_threshold = options[attributes[gear_type]["iou_threshold"]]
    iou_cost = iou_cost_matrix(person_detections, gear_detections)
    iou_cost[iou_cost > (1 - iou_threshold)] = 1
    indices = linear_assignment(iou_cost)

    for det_index, gear_index in indices:
        det = person_detections[det_index]
        best_match = gear_detections[gear_index]
        cost_iou = iou_cost[det_index][gear_index]
        if cost_iou >= 1:
            continue

        det.extra_data[attributes[gear_type]["confidence"]] = \
            best_match.extra_data["detection_confidence"]
        det.extra_data[attributes[gear_type]["iou"]] = 1 - cost_iou
        det.attributes[attributes[gear_type]["attribute_name"]] = \
            attributes[gear_type]["possible_values"][1]
```
Then in the process_frame function all you need to do is 
```python3
assign_gear_attributes(person_detections, gear_detections, "Safety Hat")
assign_gear_attributes(person_detections, gear_detections, "Safety Vest")
```

The reason I prefer this is because it leads to less nested for loops, helping with readability.